apps/trackers/product_definition_handlers/ubi_handler.py

```diff
@@ -22,12 +23,17 @@ def has_ubi_packages(ps_module: PsModule, affect: Affect) -> bool:
         return bool(packages)
 
     def get_offer(self, affect: Affect, impact: Impact, ps_module: PsModule, offers):
+        if FedrampHandler.will_modify_offers(affect, impact, ps_module):
+            # FedrampHandler is next and it would have to undo everything done by UbiHandler.
+            return offers
```

Comment:
I would be for some more general approach (considering there may be more handlers in the future) but I am not strongly against this.

Reply:
In SFM2, the function `unacked_target_allowed` does something similar to `.will_modify_offers` (given the structural differences in SFM2's vs. OSIDB's versions of the algorithm) and duplicates some conditions from the would-be handlers to predict the future and decide whether to select the unacked stream.

Reply:
Maybe just leave it as it is. I agree that there is a few handlers and it is still all pretty clear. Definitely better than in SFM2. If the number of handlers increases, we can think about something more general.

Reply:
Got it. I made some changes (basically, ubi_handler gets .will_modify_offers) that I can absolutely revert to the state you reviewed if anyone so wishes. It makes it more straightforward in my opinion, but I'm aware that most people don't share my tastes. I'll present the changes with the final review request.

Reply:
All handlers now try to predict the future (what the higher-priority == later running handlers will do) in a consistent way.